File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.py
ch = input()

a = [0] * 26
for item in ch:
  a[ord(item) - ord('A')] += 1

flag = True
if len(ch) % 2 == 0:
  for i in range(26):
    if a[i] % 2 != 0:
      flag = False
      break
  
  if flag:
    for i in range(26):
      if a[i] > 0:
        temp = a[i] // 2
        while(a[i] > temp):
          a[i] -= 1
          print(chr(i + ord('A')), end = '')
   
    for i in range(25, -1, -1):
      if a[i] > 0:
        while(a[i] > 0):
          a[i] -= 1
          print(chr(i + ord('A')), end = '')

else:
  count = 0
  for i in range(26):
    if a[i] % 2 != 0:
      c = chr(i + ord('A'))
      a[i] -= 1
      count += 1
      if count > 1:
        flag = False
        break

  if flag:
    for i in range(26):
      if a[i] > 0:
        temp = a[i] // 2
        while(a[i] > temp):
          a[i] -= 1
          print(chr(i + ord('A')), end = '')

    print(c, end = '')
   
    for i in range(25, -1, -1):
      if a[i] > 0:
        while(a[i] > 0):
          a[i] -= 1
          print(chr(i + ord('A')), end = '')



if(not flag):
  print("I'm Sorry Hansoo\n");

File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.c
#include <stdio.h>
#include <string.h>

int main(void) { 
   char str[51];
   scanf("%s", str);

   int a[26] = {0}, flag = 1, count = 0;
   for(int i = 0; i < strlen(str); i++){
      a[str[i] - 'A'] += 1;
   }
   if(strlen(str) % 2 == 0){
      for(int i = 0; i < 26; i++){
         if(a[i] % 2 != 0){
            flag = 0;
            break;
         }
      }
      if(flag){
         for(int i = 0; i < 26; i++){
            if(a[i] > 0){
               int temp = a[i] / 2;
               while(a[i] > temp){
                  a[i] -= 1;
                  printf("%c", i + 'A');
               }
            }
         }

         for(int i = 25; i >= 0; i--){
            if(a[i] > 0){
               while(a[i] > 0){
                  a[i] -= 1;
                  printf("%c", i + 'A');
               }
            }
         }
      }
   }
   else{
      char c;
      for(int i = 0; i < 26; i++){
         if(a[i] % 2 != 0){
            count += 1;
            c = i + 'A';
            a[i] -= 1;
            if(count > 1){
               flag = 0;
               break;
            }
         }
      }
      
      if(flag){
         for(int i = 0; i < 26; i++){
            if(a[i] > 0){
               int temp = a[i] / 2;
               while(a[i] > temp){
                  a[i] -= 1;
                  printf("%c", i + 'A');
               }
            }
         }

         printf("%c", c);

         for(int i = 25; i >= 0; i--){
            if(a[i] > 0){
               while(a[i] > 0){
                  a[i] -= 1;
                  printf("%c", i + 'A');
               }
            }
         }
      }
   }
   
   if(!flag)   printf("I'm Sorry Hansoo\n");
   
   return 0;
}